webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx

```diff
@@ -43,27 +50,27 @@ function useTimePostBoxIndicator(channelId: string) {
     const [timestamp, setTimestamp] = useState(0);
     const [showIt, setShowIt] = useState(false);
 
-    // get teammate timezone information
-    const teammateTimezone = useSelector(
-        (state: GlobalState) => {
-            if (!teammateId) {
-                return DEFAULT_TIMEZONE;
-            }
-
-            const teammate = getUser(state, teammateId);
-            return getTimezoneForUserProfile(teammate);
-        },
-        (a, b) =>
-            a.automaticTimezone === b.automaticTimezone &&
-            a.manualTimezone === b.manualTimezone &&
-            a.useAutomaticTimezone === b.useAutomaticTimezone,
-    );
+    const teammate: UserProfile | undefined = useSelector((state: GlobalState) => getUser(state, teammateId));
+    const teammateTimezone = useMemo(() => {
+        if (!teammate) {
+            return DEFAULT_TIMEZONE;
+        }
+
+        return getTimezoneForUserProfile(teammate);
+    }, [teammate]);
 
     // current user timezone
     const userCurrentTimezone = useSelector((state: GlobalState) => getCurrentTimezone(state));
 
     // UseEffect to update the timestamp and the visibility for the time indicator
     useEffect(() => {
+        if (isDM && teammate?.is_bot) {
```

Comment:
Should we add the self dm check here too?

Reply:
Its already there, maybe you had an old tab open?

Reply:
No, in this file there is no self dm check. It is on the other file, though.